ui/apps/platform/src/Containers/NetworkGraph/common/NetworkPolicies.tsx

```diff
@@ -18,39 +19,61 @@ import { MoonIcon, SunIcon } from '@patternfly/react-icons';
 import download from 'utils/download';
 import SelectSingle from 'Components/SelectSingle';
 import { useTheme } from 'Containers/ThemeProvider';
-import { NetworkPolicy } from 'types/networkPolicy.proto';
 import useFetchNetworkPolicies from 'hooks/useFetchNetworkPolicies';
 
 type NetworkPoliciesProps = {
+    entityName: string;
     policyIds: string[];
 };
 
-const downloadYAMLHandler = (fileName: string, fileContent: string) => () => {
-    download(`${fileName}.yml`, fileContent, 'yml');
+type NetworkPolicyYAML = {
+    name: string;
+    yaml: string;
 };
 
-function NetworkPolicies({ policyIds }: NetworkPoliciesProps): React.ReactElement {
+function NetworkPolicies({ entityName, policyIds }: NetworkPoliciesProps): React.ReactElement {
     const { networkPolicies, isLoading, error } = useFetchNetworkPolicies(policyIds);
     const { isDarkMode } = useTheme();
     const [customDarkMode, setCustomDarkMode] = React.useState(isDarkMode);
+
+    const allNetworkPoliciesYAML = useMemo(
+        () => ({
+            name: 'all',
```

Comment:
That's a good question actually. It could be possible to make one named `all`, but I'm not sure if it would be likely. Just to be safe, I also think having a more unlikely name would be beneficial. I'll try to make that change